### ISSUE TITLE
Run with --net=host break access to vnc viewer

### ISSUE THREAD

**Issue by samuelbernardolip** [solarkennedy/ipmi-kvm-docker#10]
Running docker with the following example breaks access to vnc viewer:
docker run --rm --net=host -d --name ipmi-kvm-docker --hostname ipmi-kvm-docker solarkennedy/ipmi-kvm-docker

Any suggestions how to use tunnels in localhost?

**Comment by samuelbernardolip** [comment]
Running within host break X11 and hang x11vnc. As a consequence, was possible to open novnc web interface, but without desktop to show since there is no X11:
```
supervisorctl status
X11                              FATAL      Exited too quickly (process log may have details)
chromium                         FATAL      Exited too quickly (process log may have details)
fluxbox                          FATAL      Exited too quickly (process log may have details)
novnc                            RUNNING    pid 10, uptime 0:21:36
x11vnc                           STARTING
```

**Comment by solarkennedy** [comment]
Hmm. Unless there are other things running with conflicting ports on the host, then this should be possible.
Are there any other hints in the X11 output as to why it crashed?

**Comment by samuelbernardolip** [comment]
@solarkennedy quick answer... I was already creating a new image from start since I wasn't understanding the dependencies. Also need older java 6 with ubuntu trusty 32bit for supported firefox.
I can't reproduce same issue without --net=host.
From the logs this is what I get (in loop spawning X11 that exits with error 1):
```
2021-02-02 22:24:46,789 CRIT Supervisor running as root (no user in config file)
2021-02-02 22:24:46,789 WARN Included extra file "/etc/supervisor/conf.d/supervisord.conf" during parsing
2021-02-02 22:24:46,823 INFO RPC interface 'supervisor' initialized
2021-02-02 22:24:46,823 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2021-02-02 22:24:46,823 INFO supervisord started with pid 1
2021-02-02 22:24:47,825 INFO spawned: 'X11' with pid 9
2021-02-02 22:24:47,826 INFO spawned: 'novnc' with pid 10
2021-02-02 22:24:47,828 INFO spawned: 'x11vnc' with pid 11
2021-02-02 22:24:47,829 INFO spawned: 'chromium' with pid 12
2021-02-02 22:24:47,831 INFO spawned: 'fluxbox' with pid 13
2021-02-02 22:24:47,854 INFO exited: X11 (exit status 1; not expected)
2021-02-02 22:24:47,930 INFO exited: fluxbox (exit status 1; not expected)
2021-02-02 22:24:48,090 INFO exited: chromium (exit status 1; not expected)
2021-02-02 22:24:48,093 CRIT reaped unknown pid 31)
2021-02-02 22:24:48,856 INFO spawned: 'X11' with pid 34
2021-02-02 22:24:48,857 INFO success: novnc entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2021-02-02 22:24:48,857 INFO success: x11vnc entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2021-02-02 22:24:48,857 INFO exited: x11vnc (exit status 1; not expected)
2021-02-02 22:24:48,858 INFO spawned: 'x11vnc' with pid 36
2021-02-02 22:24:48,862 INFO exited: X11 (exit status 1; not expected)
```

**Comment by virtzilla** [comment]
After a few frustrating rounds and hours trying to figure out why this would work on one of my hosts and not on another, I found that x11vnc with suffer a buffer overflow if run on a docker linux host that doesn't have ipv6 enabled (based on some googling and this https://bugzilla.redhat.com/show_bug.cgi?id=972618). 

In my case, the x11vnc crash occurs with this docker image with and without the net=host option if ipv6 is disabled on the docker host. I tried enabling ipv6 for the docker daemon itself (not on the docker host itself) and saw the same behavior with x11vnc crashing. Enabling ipv6 on the host and restarting the docker daemon resolved it.
